crates/symbolicator-service/src/caching/cache_key.rs

```diff
@@ -76,6 +58,30 @@ impl CacheKey {
         path
     }
 
+    /// Creates a stable relative path for the given [`RemoteFile`] tied to [`Scope`].
+    ///
+    /// This is being used as the "legacy" [`CacheKey`], and also forms the basis for the more
+    /// precise modern cache key.
+    pub fn relative_scoped_file(scope: &Scope, file: &RemoteFile) -> String {
+        let scope = safe_path_segment(scope.as_ref());
+        let cache_key = safe_path_segment(&file.cache_key());
+        format!("{scope}/{cache_key}")
+    }
+
+    /// Create a [`CacheKeyBuilder`] that can be used to build a cache key consisting of all its
+    /// contributing sources.
+    pub fn legacy_builder(scope: &Scope, file: &RemoteFile) -> CacheKeyBuilder {
```

Comment:
I don't understand why this is called `legacy_builder`.

Reply:
It eagerly writes the primary file source right now to populate the `legacy_cache_key`. We won’t need that later on, or rather we explicitly don’t want this for things like sourcemaps, because there is not really a "primary" file, and the cache metadata/key would look very different.